src/lib/http-server/common.ts

```diff
@@ -1,7 +1,85 @@
+import type { Account } from "@keetanetwork/keetanet-client/lib/account.js";
+import { KeetaAnchorUserError } from "../error.js";
+import { KeetaNet } from "../../client/index.js";
+import { createAssertEquals } from "typia";
+
 export interface HTTPSignedField {
 	nonce: string;
 	/* Date and time of the request in ISO 8601 format */
 	timestamp: string;
 	/* Signature of the account public key and the nonce as an ASN.1 Sequence, Base64 DER */
 	signature: string;
 }
+
+export const assertHTTPSignedField: (input: unknown) => HTTPSignedField = createAssertEquals<HTTPSignedField>();
+
+export interface HTTPSignedFieldURLParameters {
+	signedField: HTTPSignedField | null;
+	account: Account | null;
+}
+
+export function addSignatureToURL(input: URL | string, data: HTTPSignedFieldURLParameters): URL {
+	let url: URL;
+
+	if (typeof input === 'string') {
+		url = new URL(input);
+	} else {
+		url = new URL(input.toString());
+	}
+
+	for (const key of [ 'nonce', 'timestamp', 'signature' ] as const) {
+		const searchKey = `signed.${key}`;
+
+		if (url.searchParams.has(searchKey)) {
+			throw(new KeetaAnchorUserError(`URL already has signed field parameter: ${searchKey}`));
+		}
+
+		if (data.signedField) {
+			url.searchParams.set(`signed.${key}`, data.signedField[key]);
+		}
+	}
+
+	if (data.account) {
+		url.searchParams.set('account', data.account.publicKeyString.get());
+	}
+
+	return(url);
+}
+
+export function parseSignatureFromURL(input: URL | string): HTTPSignedFieldURLParameters {
```

Comment:
This should probably just be a static method on the HTTP server object rather than a part of the export object